src/Physics/Multinucleon/EventGen/MECGenerator.cxx

```diff
@@ -669,6 +669,13 @@ void MECGenerator::SelectNSVLeptonKinematics (GHepRecord * event) const
   bool accept = false;
   unsigned int iter = 0;
 
+  // Get tensor pdg and hadrontensor:
+  int tensor_pdg = fXSecModel->GetTensorPdg( TgtPDG ) ; 
```

Comment:
In order to retrieve the Tensor model, you can get the configuration of the cross section model and invoke the model via the AlgFactory.